tst/testbugfix/2012-09-06-t00253.tst

```diff
@@ -13,6 +13,6 @@ gap> Read(s);
 Syntax error: ) expected in stream:4
     v := rec(a := [];);
                     ^
-Syntax error: end expected in stream:5
+Syntax error: inside a function, statement or 'end' expected in stream:5
```

Comment:
reading this again today, perhaps a `semicolon` would make this a bit clearer, as here (note that the "expected" is added automatically at the end, so we have to contort ourselves a bit to fit the sentence structure in there)
```suggestion
Syntax error: inside a function; statement or 'end' expected in stream:5
```

Thoughts?

Reply:
While ; might be more mathematically correct, I feel that in code I always read it as "end of statement".

How about "Syntax error while parsing a function: statement or 'end' expected" (so push the new stuff before the semicolon)

Reply:
The "Syntax error:" prefix and "expected" suffix are fixed, I can only fill in stuff between them, so the best I can offer w/o deeper changes is this:
> Syntax error: while parsing a function: statement or 'end' expected

Reply:
What do you think about either of these options, which break the error into two sentences with a full stop? (It also adds commas around the "or" option; the difference between the two options is the presence of the word "was")

> Syntax error: inside a function. A statement, or 'end', was expected in stream:5

> Syntax error: inside a function. A statement, or 'end', expected in stream:5

However I don't feel particularly strongly about any of the suggestions put forth in this PR/discussion, so I don't really mind.

Reply:
Now updated

Reply:
@wilfwilson sorry, I did not actually see your comment before I updated this PR, i.e., I did not deliberately ignore it! Your variant is also fine by me. I'll think about it again tomorrow, after some sleep. Perhaps others have an opinion, too